src/pybind/mgr/dashboard/controllers/settings.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from contextlib import contextmanager
+
+import cherrypy
+
+
+from . import ApiController, RESTController
+from ..settings import Settings, Options
+from ..security import Scope
+
+
+@ApiController('/settings', Scope.CONFIG_OPT)
+class SettingsController(RESTController):
+    """
+    Enables to manage the settings of the dashboard (not the Ceph cluster).
+    """
+    @contextmanager
+    def _attribute_handler(self, name):
+        """
+        :type name: str|dict[str, str]
+        :rtype: str|dict[str, str]
+        """
+        if isinstance(name, dict):
+            result = {self._to_native(key): value
+                      for key, value in name.items()}
+        else:
+            result = self._to_native(name)
+
+        try:
+            yield result
+        except AttributeError:
+            raise cherrypy.NotFound(result)
+
+    @staticmethod
+    def _to_native(setting):
+        return setting.upper().replace('-', '_')
+
+    def list(self):
+        return [
+            self._get(name) for name in Options.__dict__
+            if name.isupper() and not name.startswith('_')
+        ]
+
+    def _get(self, name):
+        with self._attribute_handler(name) as sname:
+            default, data_type = getattr(Options, sname)
+        return {
+            'name': sname,
+            'default': default,
+            'type': data_type.__name__,
+            'value': getattr(Settings, sname)
+        }
+
+    def get(self, name):
+        return self._get(name)
+
+    def set(self, name, value):
+        with self._attribute_handler(name) as sname:
+            setattr(Settings, self._to_native(sname), value)
+
+    def delete(self, name):
+        with self._attribute_handler(name) as sname:
+            delattr(Settings, self._to_native(sname))
+
+    def bulk_set(self, **kwargs):
```

Comment:
The swagger API docs generator doesn't play well with thie `**kargs` parameter (documentation says there is not parameters):

![screenshot_20180809_160044](https://user-images.githubusercontent.com/14297426/43907636-d9bd74f8-9bed-11e8-89e1-138efd992c34.png)

Is this something that can be fixed?

Reply:
I've looked into it and yes, I think it can be fixed but it's a general issue with generating the API documentation (or how the Endpoints are generated) for the `bulk_set` method of the `RestController`. Hence, I'd say it shouldn't be fixed in this PR.

Reply:
Thanks for the info. Agree, this should be fixed in a separate PR.